textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexToGramQueryTranslatorTest.java

```diff
@@ -17,133 +17,321 @@ private void printTranslatorResult(String regex) {
+		
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
+		GramBooleanQuery expectedFirstAnd = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
+		expectedFirstAnd.operandSet.addAll(Arrays.asList(
+				new String[]{"abp", "bpq"}));
+		expectedQuery.subQuerySet.add(expectedFirstAnd);
+		GramBooleanQuery expectedSecondAnd = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
+		expectedSecondAnd.operandSet.addAll(Arrays.asList(
+				new String[]{"abc", "bcp", "cpq"}));
+		expectedQuery.subQuerySet.add(expectedSecondAnd);
+		
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));		
 	}
 	
 	@Test
 	// RE2J will simplify REPEAT to equivalent form with QUEST.
 	// abc{1,3} will be simplified to abcc?c?
 	public void testRepeat1() {
-		printTranslatorResult("abc{1,3}");
+		String regex = "abc{1,3}";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
+		GramBooleanQuery expectedAndNode = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
+		expectedAndNode.operandSet.addAll(Arrays.asList(new String[]{"abc"}));
+		expectedQuery.subQuerySet.add(expectedAndNode);
+		
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));
 	}
 	
 	@Test
 	public void testCapture1() {
-		printTranslatorResult("(abc)(qwer)");
+		String regex = "(abc)(qwer)";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
+		GramBooleanQuery expectedFirstAnd = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
+		expectedFirstAnd.operandSet.addAll(Arrays.asList(
+				new String[]{"abc", "bcq", "cqw", "qwe", "wer"}));
+		expectedQuery.subQuerySet.add(expectedFirstAnd);
+		
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));		
 	}
 	
 	@Test
 	public void testRegexCropUrl() {
-		printTranslatorResult("^(https?:\\/\\/)?([\\da-z\\.-]+)\\.([a-z\\.]{2,6})([\\/\\w \\.-]*)*\\/?$");
+		String regex = "^(https?:\\/\\/)?([\\da-z\\.-]+)\\.([a-z\\.]{2,6})([\\/\\w \\.-]*)*\\/?$";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
+		
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));
```

Comment:
a general suggestion, if you've already had the expected results, it's better to use `Assert.equal`.